contrib/gazebo/gazebo_sawyer_env.py

```diff
@@ -0,0 +1,499 @@
+
+    def _set_action_space(self):
+        # For sawyer's joints
+        # [head_pan, right_gripper_l_finger_joint, right_gripper_r_finger_joint, right_j0, right_j1,
+        # right_j2, right_j3, right_j4, right_j5, right_j6]
+        # For our use
+        # joints[3:10]
+        # set the action space depending on different control modes
+        if self._control_mode == 'position':
+            lower_bounds = np.array(self._joint_limits.position_lower[3:10])
+            upper_bounds = np.array(self._joint_limits.position_upper[3:10])
+        elif self._control_mode == 'velocity':
+            lower_bounds = np.zeros(7)
+            upper_bounds = np.array(self._joint_limits.velocity[3:10])
+        elif self._control_mode == 'effort':
+            lower_bounds = np.zeros(7)
+            upper_bounds = np.array(self._joint_limits.effort[3:10])
+        else:
+            # should raise an unknown mode exception
+            raise ControlModeUnknownErr(
+                "Control mode %s is not known!" % self._control_mode)
+
+        self._action_space = Box(
+            np.concatenate((lower_bounds, np.array([0]))),
+            np.concatenate((upper_bounds, np.array([100]))))
+
+    def update_model_states(self, data):
+        self.model_states = data
+
+    def compute_reward(self, achieved_goal, goal, info=None):
+        # Compute distance between goal and the achieved goal.
+        d = goal_distance(achieved_goal, goal)
+        if self.reward_type == 'sparse':
+            return -(d > self.distance_threshold).astype(np.float32)
+        else:
+            return -d
+
+    # Implementation for rllab env functions
+    # -----------------------------------
+    @rate_limited(100)
+    def step(self, action):
+        """
+        Perform a step in gazebo. When end of episode
+        is reached, reset() should be called to reset the environment's internal state.
+        Input
+        -----
+        action : an action provided by the environment
+        Outputs
+        -------
+        (observation, reward, done, info)
+        observation : agent's observation of the current environment
+        reward [Float] : amount of reward due to the previous action
+        done : a boolean, indicating whether the episode has ended
+        info : a dictionary containing other diagnostic information from the previous action
+        """
+        print("Step forwarding...")
+
+        if self._control_mode != 'position':
+            rospy.wait_for_service('/gazebo/unpause_physics')
+            try:
+                unpause = rospy.ServiceProxy('/gazebo/unpause_physics',
+                                             std_srvs.srv.Empty)
+                unpause()
+            except rospy.ServiceException as e:
+                print('/gazebo/unpause_physics service call failed')
+
+        # set joint angles
+        joint_angle_cmds = {
+            'right_j0': action[0],
```

Comment:
```python
joint_angle_cmds = {}
for i in range(7):
    joint_angle_cmds['right_j{}'.format(i)'] = action[i]
```

also, if it is _always_ right_j[n], how is there a parameter for which limb to use?